lib/device-log/safari-network-log.js

```diff
@@ -0,0 +1,140 @@
+import _ from 'lodash';
+import URL from 'url';
+import { RotatingLog, MAX_LOG_ENTRIES_COUNT } from './rotating-log';
+
+
+class SafariNetworkLog extends RotatingLog {
+  constructor (showLogs) {
+    super(showLogs, 'SafariNetwork');
+  }
+
+  getEntry (requestId) {
+    let outputEntry;
+    while (this.logs.length >= MAX_LOG_ENTRIES_COUNT) {
+      // pull the first entry, which is the oldest
+      const entry = this.logs.shift();
+      if (entry && entry.requestId === requestId) {
+        // we are adding to an existing entry, and it was almost removed
+        // add to the end of the list and try again
+        outputEntry = entry;
+        this.logs.push(outputEntry);
+        continue;
+      }
+      // we've removed an element, so the count is down one
+      if (this.logIdxSinceLastRequest > 0) {
+        this.logIdxSinceLastRequest--;
+      }
+    }
+
+
+    if (!outputEntry) {
+      // we do not yes have an entry to associate this bit of output with
+      // most likely the entry will be at the end of the list, so start there
+      for (let i = this.logs.length - 1; i >= 0; i--) {
+        if (this.logs[i].requestId === requestId) {
+          // found it!
+          outputEntry = this.logs[i];
+          // this is now the most current entry, so remove it from the list
+          // to be added to the end below
+          this.logs.splice(i, 1);
+          break;
+        }
+      }
+
+      // nothing has been found, so create a new entry
+      if (!outputEntry) {
+        outputEntry = {
+          requestId,
+          logs: [],
+        };
+      }
+
+      // finally, add the entry to the end of the list
+      this.logs.push(outputEntry);
+    }
+
+    return outputEntry;
+  }
+
+  addLogLine (method, out) {
+    if (['Network.dataReceived'].includes(method)) {
+      // status update, no need to handle
+      return;
+    }
+
+    // events we care about:
+    //   Network.requestWillBeSent
+    //   Network.responseReceived
+    //   Network.loadingFinished
+
+    const outputEntry = this.getEntry(out.requestId);
+    if (this.isCapturing) {
+      // now add the output we just received to the logs for this particular entry
+      outputEntry.logs = outputEntry.logs || [];
```

Comment:
do we want to limit the count of log items in each entry?

Reply:
There will only ever be three.